server/src/test/java/org/elasticsearch/index/engine/NoopEngineTests.java

```diff
@@ -56,4 +67,38 @@ public void testTwoNoopEngines() throws IOException {
         engine2.close();
     }
 
+    public void testNoopAfterRegularEngine() throws IOException {
+        int docs = randomIntBetween(1, 10);
+        ReplicationTracker tracker = (ReplicationTracker) engine.config().getGlobalCheckpointSupplier();
```

Comment:
why is this needed? we typically just write to the engine?

Reply:
This is a rabbit hole @jasontedor and I got into last night; in order to completely flush and remove the translog we need a deletion policy where we can advance the generation needed for recovery, in order to advance that we have to have the tracker working correctly (which I guess is normally managed at the IndexShard level), so this allows us to advance global checkpoints as needed so we can remove the translog and open a new noop engine (otherwise the noop engine would throw an exception about there being translog operations when being opened).